app/src/main/java/com/example/pocketcloset/Fragments/ProfileFragment.java

```diff
@@ -121,4 +274,175 @@ public void onActivityResult(int requestCode, int resultCode, @Nullable Intent d
             }
         }
     }
+
+    public List<Clothing> filterClothing(List<Clothing> allClothing, String clothingType) {
+        ArrayList<Clothing> newClothing = new ArrayList<>();
+        for (int i = 0; i < allClothing.size(); i++) {
+            if (allClothing.get(i).getClothingType().equals(clothingType)) {
+                newClothing.add(allClothing.get(i));
+            }
+        }
+        return newClothing;
+    }
+
+    public Clothing MostWorn(String clothingType) {
+        List<Clothing> clothing = filterClothing(allClothing, clothingType);
+        if(clothing != null) {
+            Collections.sort(clothing, (d1, d2) -> {
```

Comment:
I think this can be accomplished as part of the Parse query. Can you order the results by worn count? Then if you filter by clothing type in the query too, this function won't be necessary :)

Reply:
Could you clarify more on this? If I order results by worn count, how do I parse the least or most worn clothing from each type of article?